ext/pdo_firebird/tests/fb4_datatypes.phpt

```diff
@@ -0,0 +1,52 @@
+--TEST--
+PDO_Firebird: Supported Firebird 4.0 datatypes
+--EXTENSIONS--
+pdo_firebird
+--SKIPIF--
+<?php require('skipif.inc'); ?>
```

Comment:
Versions lower than 4 may want to skip this test

Reply:
How to set this restriction?

```php
$dbh->getAttribute(PDO::ATTR_SERVER_VERSION);
```

Gets the server version, but it looks like this

```
Firebird/Linux/AMD/Intel/x64 (access method), version "LI-V5.0.1.1391 Firebird 5.0 faca113" Firebird/Linux/AMD/Intel/x64 (remote server), version "LI-V5.0.1.1391 Firebird 5.0 faca113/tcp (DESKTOP-E3INAFT)/P18:C" Firebird/Linux/AMD/Intel/x64 (remote interface), version "LI-V5.0.1.1391 Firebird 5.0 faca113/tcp (DESKTOP-E3INAFT)/P18:C" on disk structure version 13.1
```

You also need to use a regular expression to make this information simple.

On the other hand, the engine version can be obtained by request

```sql
select rdb$get_context('SYSTEM', 'ENGINE_VERSION')
from rdb$database
```

There is only a version number. But in any case, the server version can only be found after connecting to the database.

Create a separate function that returns the engine version?

Reply:
What matters here is the client version, not the server version. So, could you please try the following constant?

```
PDO::ATTR_CLIENT_VERSION
```

Reply:
It also returns a complex version of the client

LI-V6.3.1.1391 Firebird 5.0

The success of this test depends less on the client (although it does too) and more on the Firebird header files.

```cpp
#if FB_API_VER >= 40
...
#endif
```

FB_API_VER is defined in ibase.h, and if the header file is from Firebird 3.0, this branch simply will not end up in the binary file when compiled.

Reply:
Yes, sure. The API version is essentially the version of the header files. And the client version is the version of `fbclient.dll` or `libfbclient.so`. They are generally independent.

What is the point of the patch? After preparing query, the server says I have these types of columns and input parameters. We cannot process some of them, so in response we tell the server, instead of these types, return VARCHAR(N) to us. And this part, in theory, can generally work independently of the client (I have not tested).

> Now, probably the simplest solution here is to be able to retrieve FB_API_VER. In fact, this is useful for users when debugging on their own.

> Could you please wait a moment while I open the PR to add?

OK. I also think this will be useful.

Reply:
Opened PR 
https://github.com/php/php-src/pull/14916